code/modules/reagents/reagent_containers/iv_bag.dm

```diff
@@ -0,0 +1,179 @@
+				injection_target.reagents.trans_to(src, amount_per_transfer_from_this/10)
+			update_icon()
+
+/obj/item/reagent_containers/iv_bag/attack(mob/living/M, mob/living/user, def_zone)
+	return
+
+/obj/item/reagent_containers/iv_bag/afterattack(atom/target, mob/user, proximity)
+	if(!proximity)
+		return
+	if(!target.reagents)
+		return
+
+	if(isliving(target))
+		var/mob/living/L = target
+		if(injection_target) // Removing the needle
+			if(L != injection_target)
+				to_chat(user, "<span class='notice'>[src] is already inserted into [injection_target]'s arm!")
+				return
+			if(L != user)
+				L.visible_message("<span class='danger'>[user] is trying to remove [src]'s needle from [L]'s arm!</span>", \
+								"<span class='userdanger'>[user] is trying to remove [src]'s needle from [L]'s arm!</span>")
+				if(!do_mob(user, L))
+					return
+			L.visible_message("<span class='danger'>[user] removes [src]'s needle from [L]'s arm!</span>", \
+								"<span class='userdanger'>[user] removes [src]'s needle from [L]'s arm!</span>")
+			end_processing()
+		else // Inserting the needle
+			if(!L.can_inject(user, 1))
+				return
+			if(L != user)
+				L.visible_message("<span class='danger'>[user] is trying to insert [src]'s needle into [L]'s arm!</span>", \
+									"<span class='userdanger'>[user] is trying to insert [src]'s needle into [L]'s arm!</span>")
+				if(!do_mob(user, L))
+					return
+			L.visible_message("<span class='danger'>[user] inserts [src]'s needle into [L]'s arm!</span>", \
+									"<span class='userdanger'>[user] inserts [src]'s needle into [L]'s arm!</span>")
+			begin_processing(L)
+
+
+/obj/item/reagent_containers/iv_bag/update_icon()
+	overlays.Cut()
+
+	if(reagents.total_volume)
+		var/percent = round((reagents.total_volume / volume) * 10) // We round the 1's place off of our percent for easy image processing.
+		var/image/filling = image('icons/goonstation/objects/iv.dmi', src, "[icon_state][percent]")
+
+		filling.icon += mix_color_from_reagents(reagents.reagent_list)
+		overlays += filling
+	if(ismob(loc))
+		switch(mode)
+			if(IV_DRAW)
+				overlays += "draw"
+			if(IV_INJECT)
+				overlays += "inject"
+
+/obj/item/reagent_containers/iv_bag/proc/set_label(new_label)
+	name = "[initial(name)] ([new_label])"
+
+/obj/item/reagent_containers/iv_bag/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/pen) || istype(I, /obj/item/flashlight/pen))
+		var/tmp_label = sanitize(input(user, "Enter a label for [name]","Label",label_text))
+		if(length(tmp_label) > MAX_NAME_LEN)
+			to_chat(user, "<span class='warning'>The label can be at most [MAX_NAME_LEN] characters long.</span>")
+		else
+			to_chat(user, "<span class='notice'>You set the label to \"[tmp_label]\".</span>")
+			set_label(tmp_label)
+
+// PRE-FILLED IV BAGS BELOW
+
+/obj/item/reagent_containers/iv_bag/salglu/New()
+	..()
+	set_label("Saline-Glucose")
+	reagents = list_reagents = list("salglu_solution" = 200)
```

Comment:
this isn't how this works; this needs to be on the defined type; using it after `New` is not going to do anything.